File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    finalArray = []
    for i in range(len(numbers)):
        num = list(str(numbers[i]))
        sum = 0
        for j in num:
            sum += pow(int(j), len(num))
        if sum == int(i):
            finalArray.append(sum)
    return finalArray

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function (anArr) {
  let resultArr = [];
  for (let i of anArr) {
    if (String(i).length === 1) {
      resultArr.push(i);
    } else {
      let splittedNum = String(i).split("");
      let len = splittedNum.length;
      let sum = 0;
      for (let digit of splittedNum) {
        sum += Number(digit) ** len;
      }
      if (sum === i) {
        resultArr.push(i);
      }
    }
  }
  return resultArr;
};
